### PR TITLE
chore(`ci/litjob`): fix and optimize GPU related tests

### PR DESCRIPTION
**Pull request by bhimrazy**
## What does this PR do?

This PR has the following changes: 

- [x] Reduced GPU test time by approximately **25–30** minutes.  
   Two of these tests were taking more than 20 minutes but, when run manually, finished within 5 seconds.  
   - test_tensorrt_save_ir_type[ts-ScriptModule]
   - test_tensorrt_export_reload[ts-torchscript]
  <a href="https://lightning.ai/lightning-ai/ci/jobs/ci-run-lightning-ai-pytorch-lightning-573828a-pytorch-yml-lit-job-pytorch-3-12-1578b15a?app_id=jobs&job_detail_tab=logs">image</a>

    
    The delay in those tests was mainly caused by `CUDA_LAUNCH_BLOCKING=1`.

    **Before:**
    image
    **After:**
    image

- [x] Fixes a typo by replacing `$(PACKAGE_NAME)` with `${PACKAGE_NAME}`.
- [x] Fix: `unzip` not found — used in  ci
    ```
    .actions/pull_legacy_checkpoints.sh: line 10: unzip: command not found
    total 0
    + cd tests/legacy
    + ls -lh checkpoints/
    total 0
    ```
    https://github.com/Lightning-AI/pytorch-lightning/blob/573828a19fcb78c7871946abf8e750256769a120/.actions/pull_legacy_checkpoints.sh#L10

**Note:** It seems there are still `ci/litjob` failures caused by port issues, which require a rerun to pass.

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Comment by bhimrazy** [comment]
It seems the delay in those tests is caused by `CUDA_LAUNCH_BLOCKING=1`.
<details><summary>Test Code</summary>
<p>

```python
export CUDA_LAUNCH_BLOCKING=0
time python -c "
import torch
from lightning.pytorch.demos.boring_classes import BoringModel
model = BoringModel()
model.example_input_array = torch.randn((4, 32))
model.cuda()
print('Starting compilation with CI env vars...')
import time
start = time.time()
ret = model.to_tensorrt(ir='ts')
end = time.time()
print(f'Compilation took {end - start:.2f} seconds')
"
```
</p>
</details> 

> Almost 20 mins saved

File: .lightning/workflows/pytorch.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ env:
   DEBIAN_FRONTEND: "noninteractive"
   CUDA_TOOLKIT_ROOT_DIR: "/usr/local/cuda"
   MKL_THREADING_LAYER: "GNU"
-  CUDA_LAUNCH_BLOCKING: "1"
+  # CUDA_LAUNCH_BLOCKING: "1" # disabling to speed up tests
```
Comment (bhimrazy) [inline .lightning/workflows/pytorch.yml]:
```suggestion

```

**Review by Borda — APPROVED** [review]
